fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -1013,6 +1013,14 @@ private CompletableFuture<Integer> moveDocsFromPartition(@Nonnull final LuceneRe
                 newBoundaryPartitionKey = toPartitionKey(records.get(records.size() - 1));
                 records.remove(records.size() - 1);
             }
+
+            if (records.size() == 0) {
```

Comment:
Should this be above the check above 
```
if (!repartitioningContext.emptyingPartition && repartitioningContext.newBoundaryRecordPresent) {
```

Reply:
`records.size()` should be `> 0` in all cases **unless** `fetchedRecordsFuture` failed to fetch the required records. The check here is made to ensure that `records.remove(records.size() - 1)` a couple of lines above didn't empty the records list.